### DISCUSSION TITLE
Disposing capsule

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#158] [General]
When I dispose a Capsule by implementing a custom Handle, what happens to another Capsule that it using that capsule? Does it get disposed too? Nothing happens?

**Comment by GregoryConrad** [comment]
I'm having a bit of a hard time following exactly what you mean; could you provide a few line code example?

**Comment by busslina** [comment]
I meant, what is the behaviour on `CapsuleA` when `capsuleB` get disposed? But I suppose nothing happens. In other words, it doesn't get notified that `CapsuleB` has been disposed.

```dart
int capsuleA(CapsuleHandle use) {
  final otherCapsuleValue = use(capsuleB);
}
```

**Reply by busslina** [reply]
BTW, regarding the web framework, I'm following a similar Flutter structure, with an `Element` tree and Components. I didn't get a whole of the idea you proposed. I found useful the dynamic capsules. And maybe over the time I achieve a more ReArch fine integration there

**Reply by GregoryConrad** [reply]
> I meant, what is the behaviour on CapsuleA when capsuleB get disposed?

In your provided example, `capsuleA` is a dependent of `capsuleB`. Thus, when `capsuleB` is disposed (which, at the moment, is only possible if it and all of its transitive dependents are "idempotent"), then `capsuleA` is also disposed too.

**Reply by busslina** [reply]
Well, I was thinking in the context of managed capsules with a custom ComponentHandle. Would be enough to dispose a Component in order to dispose all the descendants?

**Reply by GregoryConrad** [reply]
It depends on a few things, but if you force dispose a component (or capsule), all dependents would need to be disposed too
